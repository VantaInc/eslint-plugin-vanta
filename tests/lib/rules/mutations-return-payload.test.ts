import { GraphQLRuleTester } from "@graphql-eslint/eslint-plugin";
import rule from "../../../lib/rules/mutations-return-payload";

const ruleTester = new GraphQLRuleTester();

ruleTester.runGraphQLTests("mutations-return-payload", rule, {
  valid: [
    {
      code: `
type Mutation {
  validMutation: abcPayload
  anotherOne: bcdPayload
}
`,
    },
    {
      code: `
type Mutation {
  validMutation: abcPayload
}

extend type Mutation {
  anotherOne: bcdPayload
}
`,
    },
  ],
  invalid: [
    {
      code: `
type Mutation {
  invalid: bcd
}
`,
      errors: 1,
    },
    {
      code: `
type Mutation {
  invalid: bcd!
}
`,
      errors: 1,
    },
    {
      code: `
type Mutation {
  invalid: [bcdPayload!]!
}
`,
      errors: 1,
    },
    {
      code: `
type Mutation {
  invalid: [bcdPayload]
}
`,
      errors: 1,
    },
    {
      code: `
type Mutation {
  validMutation: abcPayload
}

extend type Mutation {
  invalidOne: bcd
}
`,
      errors: 1,
    },
    {
      code: `
type Mutation {
  invalidMutation: abcPayload!
}

extend type Mutation {
  another: bcd
}
`,
      errors: 2,
    },
  ],
});
