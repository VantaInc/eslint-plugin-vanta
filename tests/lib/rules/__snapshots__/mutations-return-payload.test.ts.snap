// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid #1 1`] = `
"#### ⌨️ Code

      1 | type Mutation {
      2 |   invalid: bcd
      3 | }

#### ❌ Error

    > 1 | type Mutation {
        | ^^^^^^^^^^^^^^^
    > 2 |   invalid: bcd
        | ^^^^^^^^^^^^^^
    > 3 | }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Mutations must return types with the suffix "Payload""
`;

exports[`Invalid #2 1`] = `
"#### ⌨️ Code

      1 | type Mutation {
      2 |   invalid: bcd!
      3 | }

#### ❌ Error

    > 1 | type Mutation {
        | ^^^^^^^^^^^^^^^
    > 2 |   invalid: bcd!
        | ^^^^^^^^^^^^^^^
    > 3 | }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Mutation payloads must be nullable"
`;

exports[`Invalid #3 1`] = `
"#### ⌨️ Code

      1 | type Mutation {
      2 |   invalid: [bcdPayload!]!
      3 | }

#### ❌ Error

    > 1 | type Mutation {
        | ^^^^^^^^^^^^^^^
    > 2 |   invalid: [bcdPayload!]!
        | ^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 | }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Mutation payloads must be nullable"
`;

exports[`Invalid #4 1`] = `
"#### ⌨️ Code

      1 | type Mutation {
      2 |   invalid: [bcdPayload]
      3 | }

#### ❌ Error

    > 1 | type Mutation {
        | ^^^^^^^^^^^^^^^
    > 2 |   invalid: [bcdPayload]
        | ^^^^^^^^^^^^^^^^^^^^^^^
    > 3 | }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Mutation payloads must not be list types"
`;

exports[`Invalid #5 1`] = `
"#### ⌨️ Code

      1 | type Mutation {
      2 |   validMutation: abcPayload
      3 | }
      4 |
      5 | extend type Mutation {
      6 |   invalidOne: bcd
      7 | }

#### ❌ Error

      4 |
    > 5 | extend type Mutation {
        | ^^^^^^^^^^^^^^^^^^^^^^
    > 6 |   invalidOne: bcd
        | ^^^^^^^^^^^^^^^^^
    > 7 | }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Mutations must return types with the suffix "Payload""
`;

exports[`Invalid #6 1`] = `
"#### ⌨️ Code

      1 | type Mutation {
      2 |   invalidMutation: abcPayload!
      3 | }
      4 |
      5 | extend type Mutation {
      6 |   another: bcd
      7 | }

#### ❌ Error 1/2

    > 1 | type Mutation {
        | ^^^^^^^^^^^^^^^
    > 2 |   invalidMutation: abcPayload!
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 | }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Mutation payloads must be nullable
      4 |

#### ❌ Error 2/2

      4 |
    > 5 | extend type Mutation {
        | ^^^^^^^^^^^^^^^^^^^^^^
    > 6 |   another: bcd
        | ^^^^^^^^^^^^^^
    > 7 | }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Mutations must return types with the suffix "Payload""
`;
